src/config/Config.zig

```diff
@@ -683,6 +683,31 @@ foreground: Color = .{ .r = 0xFF, .g = 0xFF, .b = 0xFF },
 /// on the same selection.
 @"selection-clear-on-copy": bool = false,
 
+/// Characters that mark word boundaries during text selection operations such
+/// as double-clicking. When selecting a word, the selection will stop at any
+/// of these characters.
+///
+/// This is similar to the `WORDCHARS` environment variable in zsh, except this
+/// specifies the boundary characters rather than the word characters. The
+/// default includes common delimiters and punctuation that typically separate
+/// words in code and prose.
+///
+/// Each character in this string becomes a word boundary. Multi-byte UTF-8
+/// characters are supported.
+///
+/// The null character (U+0000) is always treated as a boundary and does not
+/// need to be included in this configuration.
+///
+/// Default: ` \t'"│`|:;,()[]{}<>$`
+///
+/// To add or remove specific characters, you can set this to a custom value.
+/// For example, to treat semicolons as part of words:
+///
+///     selection-word-chars = " \t'\"│`|:,()[]{}<>$"
+///
+/// Available since: 1.2.0
+@"selection-word-chars": []const u8 = " \t'\"│`|:;,()[]{}<>$",
```

Comment:
Instead of making this a `[]const u8`, I'd recommend making a new type here that automatically expands these into a list of codepoints. 

This way we don't need an arbitrary max, we can limit it by the allocator (or put a really high limit), and we can allocate, in general! 

It also limits the runtime cost when we actually do selection since the boundary characters are already built up.

Reply:
@mitchellh, I made the change, thanks for your feedback.

Also, I ran `zig build run` and did a quick test with "some-hyphenated-words" and it worked.